src/main/scala/com/fulcrumgenomics/vcf/filtration/FilterSomaticVcf.scala

```diff
@@ -138,6 +138,34 @@ class FilterSomaticVcf
     bamIn.safelyClose()
   }
 
+  /**
+    * Filters the pileup to remove entries that are from PE reads who's insert size indicates that the
+    * aligned base we're looking at is _outside_ the insert!
+    */
+  private def filterPileup[A <: PileupEntry](pileup: Pileup[A]): Pileup[A] = {
+    val filtered = pileup.pile.filter { e =>
+      if (Bams.isFrPair(e.rec)) {
+        val (start, end) = Bams.insertCoordinates(e.rec)
+        if (pileup.pos < start || pileup.pos > end) {
+          logger.warning(
+            s"""
+              |Ignoring read ${e.rec.getReadName} mapped over variant site ${pileup.refName}:${pileup.pos} that has
+              |incompatible insert size yielding insert coordinates of ${pileup.refName}:$start-$end which do not overlap the variant.
+            """.stripMargin.trim.lines.mkString(" "))
+          false
+        }
+        else {
+          true
+        }
+      }
+      else {
+        true
```

Comment:
If it were false we'd lose all non-FR reads, which might not be desirable.  I have a hard time reasoning about what the expectations are for non-FR reads, since if the insert size if 5' to 5', then all other pair orientations will by definition have bases outside of the region defined by the insert size.